packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -256,37 +256,43 @@ class TestCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts {
 
   @override
   Future<FlutterCommandResult> verifyThenRunCommand(String? commandPath) {
-    _testFiles = argResults!.rest.map<String>(globals.fs.path.absolute).toList();
-    if (_testFiles.isEmpty) {
+    final List<Uri> testUris = argResults!.rest.map(_parseTestArgument).toList();
+    if (testUris.isEmpty) {
       // We don't scan the entire package, only the test/ subdirectory, so that
       // files with names like "hit_test.dart" don't get run.
       final Directory testDir = globals.fs.directory('test');
       if (!testDir.existsSync()) {
         throwToolExit('Test directory "${testDir.path}" not found.');
       }
-      _testFiles = _findTests(testDir).toList();
-      if (_testFiles.isEmpty) {
+      _testFileUris..clear()..addAll(_findTests(testDir).map(Uri.file));
```

Comment:
why do we have to clear `_testFileUris`?

Reply:
We shouldn't. I guess I did this to preserve the original semantics (the original `_testFiles` was overwritten). Unless this `verifyThenRunCommand` method is being called multiple times on a single instance, it shouldn't make a difference.

(I've removed it now).